test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -152,21 +154,55 @@ public void CreateBindingMetadata_EditableAttributeTrue_SetsReadOnlyFalse()
             Assert.False(context.BindingMetadata.IsReadOnly);
         }
 
+        // This is IMPORTANT. Product code needs to use GetName() instead of .Name. It's easy to regress.
+        [Fact]
+        public void CreateDisplayMetadata_DisplayAttribute_NameFromResources_NullLocalizer()
+        {
+            // Arrange
+            var provider = new DataAnnotationsMetadataProvider(stringLocalizerFactory: null);
+
+            var display = new DisplayAttribute()
+            {
+#if USE_REAL_RESOURCES
+                Name = nameof(Test.Resources.DisplayAttribute_Name),
+                ResourceType = typeof(Test.Resources),
+#else
+                Name = nameof(Test.Resources.DisplayAttribute_Name),
+                ResourceType = typeof(TestResources),
+#endif
```

Comment:
Yup. There's a pattern w.r.t. `USE_REAL_RESOURCES`. Stick to it even though the `Name` on the `#else` side would correctly be `nameof(TestResources.DisplayAttribute_Name)`. This PR isn't the place to see whether `USE_REAL_RESOURCES` is needed nor to clean up what it changes.